pco_c/include/cpcodec.h

```diff
@@ -0,0 +1,24 @@
+typedef enum PcoType {
+  U8,
+  I32,
+  I64,
+  U32,
+  U64,
+  F32,
+  F64,
+} PcoType;
+
+typedef struct PcoVec {
+  const void *ptr;
+  unsigned int len;
+  const void *raw_box;
+} PcoVec;
+
+struct PcoVec auto_compress(const void *nums,
```

Comment:
Let's not address this now, but I'd like to point out that a more C-style approach to this API would be to pass in a `char *` that pco writes data to. We'd need some way to inform C how many bytes need to be allocated, though.

Reply:
LZ4 and Zstd for example provide a function where you give it the size of the input buffer and it gives you the worst case size of the output buffer.

In LZ4 the code for this is just `(isize) + ((isize)/255) + 16)`

Reply:
Yep. I think blosc just does `(size + 16)`. Somewhat different, but we have size hints; pro: the size is more conservative and tailored to the data; con: requires 3 steps (precompute stuff, get conservative size estimate, do compression). I should be able to come up with an a-priori later though formula.